src/hooks/core/chat/useChatUtilsHook.ts

```diff
@@ -256,11 +255,19 @@ function useChatUtilsHook(): UseChatUtilsHook {
   };
 
   const goBackFromChatScreen = async () => {
-    navigation.goBack();
-    setChat((prevChat) => ({
-      ...prevChat,
-      selectedChannel: null
-    }));
+    if (type === 'ANONYMOUS') {
+      navigation.navigate('AnonymousChannelList');
+    }
+    if (type === 'SIGNED') {
+      navigation.navigate('SignedChannelList');
+    }
+    // navigation.goBack();
+    setChat((prevChat) => {
+      return {
+        ...prevChat,
+        selectedChannel: null
+      };
+    });
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The logic in `goBackFromChatScreen` has been updated to navigate to different screens based on the `type` parameter. This is a good change as it makes the navigation more context-aware. However, there's no handling for the case when `type` is neither 'ANONYMOUS' nor 'SIGNED'. It would be better to add an else clause to handle this scenario.
```diff
  const goBackFromChatScreen = async () => {
    if (type === 'ANONYMOUS') {
      navigation.navigate('AnonymousChannelList');
    }
    else if (type === 'SIGNED') {
      navigation.navigate('SignedChannelList');
    }
+   else {
+     // Handle the case when type is neither 'ANONYMOUS' nor 'SIGNED'
+     console.error(`Invalid type: ${type}`);
+   }
    setChat((prevChat) => {
      return {
        ...prevChat,
        selectedChannel: null
      };
    });
  };
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->